### PR TITLE
more test coverage for gossipsub

### PR DESCRIPTION
**Pull request by vyzo**
Adds more tests to increase coverage, as the title suggests.
See #287

### CODE REVIEW

File: gossipsub.go [hunk 1]
```diff
@@ -287,6 +287,8 @@ func (gs *GossipSubRouter) Attach(p *PubSub) {
 	// connect to direct peers
 	if len(gs.direct) > 0 {
 		go func() {
+			// add a small delay to make this unit-testable
+			time.Sleep(time.Second)
```
Comment (yusefnapora) [inline gossipsub.go]: Should we make this a variable, so that we only add the delay when testing?

**Review by yusefnapora — APPROVED** [review]
This looks great to me

File: gossipsub.go (hunk 1 above)
Comment (vyzo) [reply gossipsub.go]: Maybe so, but a 1s delay to connect can't hurt the real use case :)

Comment (vyzo) [reply gossipsub.go]: alright, I'll turn it into a variable -- not sure we gain much by it though, I'll leave it to its default value of 1s.

Comment (vyzo) [reply gossipsub.go]: done (will push shortly).

**Review by dirkmc — APPROVED** [review]

File: .travis.yml [hunk 2]
```diff
@@ -9,7 +9,6 @@ go:
 env:
   matrix:
     - GOTFLAGS="-race"
-    - GOTFLAGS="-race -tags=openssl"
```
Comment (Stebalien) [inline .travis.yml]: "fails with no output produced" is a reason to track down the bug and fix it. Let's not just remove a test.

Comment (vyzo) [reply .travis.yml]: I have added a `travis_wait` and increased the test timeout, and things seem to be smooth now.
I can restore.

Comment (vyzo) [reply .travis.yml]: I suspect the reason is that travis takes a very long time to build the tests for running, which is exacerbated with openssl.

Comment (vyzo) [reply .travis.yml]: restored.

**Pull request merged**